src/response.rs

```diff
@@ -548,39 +548,34 @@ impl Response {
         })
     }
 
-    /// Create a response from a Read trait impl.
+    /// Create a response from a DeadlineStream, reading and parsing only the status line, headers
+    /// and its following CRLF.
     ///
-    /// This is hopefully useful for unit tests.
-    ///
-    /// Example:
-    ///
-    /// use std::io::Cursor;
-    ///
-    /// let text = "HTTP/1.1 401 Authorization Required\r\n\r\nPlease log in\n";
-    /// let read = Cursor::new(text.to_string().into_bytes());
-    /// let resp = ureq::Response::do_from_read(read);
-    ///
-    /// assert_eq!(resp.status(), 401);
-    pub(crate) fn do_from_stream(stream: Stream, unit: Unit) -> Result<Response, Error> {
-        let remote_addr = stream.remote_addr;
-
-        let local_addr = match stream.socket() {
+    /// Since this function only reads the status line, header and the following CRLF, the returned
+    /// Response will have an empty reader and does not take ownership of DeadlineStream.
+    /// To read the following data, the DeadlineStream can be read again after the call to this
+    /// function.
+    /// If consume is set to true, the status line and headers will be consumed in the
+    /// DeadlineStream, so next call to read will start after the headers and their CRLF.
+    pub(crate) fn do_head_from_stream(stream: &mut DeadlineStream, unit: &Unit, consume: bool) -> Result<Response, Error> {
+        let mut tot_bytes_read = 0;
+        let remote_addr = stream.inner_ref().remote_addr;
+
+        let local_addr = match stream.inner_ref().socket() {
             Some(sock) => sock.local_addr().map_err(Error::from)?,
             None => std::net::SocketAddrV4::new(std::net::Ipv4Addr::new(127, 0, 0, 1), 0).into(),
         };
 
-        //
-        // HTTP/1.1 200 OK\r\n
-        let mut stream = stream::DeadlineStream::new(stream, unit.deadline);
-
         // The status line we can ignore non-utf8 chars and parse as_str_lossy().
-        let status_line = read_next_line(&mut stream, "the status line")?.into_string_lossy();
+        let (bytes_read, status_line) = read_next_line(stream, "the status line")?;
```

Comment:
Would it maybe be better to have `read_next_line` take a third argument `running_total: &mut usize`, so we don't need to return and remember to add to the total?